wdl-analysis/src/types/v1.rs

```diff
@@ -1432,14 +1443,47 @@ impl<'a, C: EvaluationContext> ExprTypeEvaluator<'a, C> {
                 // Evaluate the argument expressions
                 let mut count = 0;
                 let mut arguments = [const { Type::Union }; MAX_PARAMETERS];
+                let mut arguments_node = Vec::with_capacity(expr.arguments().count());
+
                 for arg in expr.arguments() {
+                    arguments_node.push(arg.clone());
                     if count < MAX_PARAMETERS {
-                        arguments[count] = self.evaluate_expr(&arg)?;
+                        arguments[count] = self.evaluate_expr(&arg).unwrap_or(Type::Union);
```

Comment:
Thanks for catching this; we should definitely be treating the argument as `Union`.

Reply:
So I think the panics in the tests are due to this change, where we're now binding to a function even if we can't evaluate one of its arguments.

I think we'll need to change `check_unnecessary_call` to accept things if the argument's type is `Union`.